src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -856,6 +856,17 @@ til::size NonClientIslandWindow::GetTotalNonClientExclusiveSize(UINT dpi) const
     };
 }
 
+// Method Description:
+// - Gets the height of title bar in physical pixels.
+// Return Value
+// - The height of title bar in physical pixels.
+[[nodiscard]] float NonClientIslandWindow::GetTitleBarPhysicalRenderHeight() const noexcept

```

Comment:
It seems a little odd to me that we need to expose this, just so that the app host can ask us for it, then tell us to do something again.

Shouldn't `IslandWindow::OpenSystemMenu` be able to figure out on its own:
* if we're maximized
* if we should correct the menu position for the NCIW (maybe using some `protected` helper)?

Reply:
It is a bit odd. Let me take another stab at it with the next iteration.